pkg/controller/plan/adapter/openstack/client.go

```diff
@@ -940,7 +948,13 @@ func (r *Client) ensureImageUpToDate(vm *libclient.VM, image *libclient.Image, v
 			r.Log.Info("the image does not exist in the inventory, waiting...",
 				"vm", vm.Name, "image", image.Name, "properties", image.Properties)
 		}
-		return
+		return false, nil
```

Comment:
this change looks incorrect since we do want to return an error when it's not of type NotFoundError, right? let's minimize the changes here so we can quickly backport it to release-v2.6.0 and post cosmetic changes separately?